### ISSUE TITLE
Add a lower limit for thickness of strokes

### ISSUE THREAD

**Issue by 4VRDriver** [xournalpp/xournalpp#2118]
**Affects versions :**
 - OS: Manjaro Linux
 - (Linux only) Desktop environment: KDE Plasma, X11
 - Which version of libgtk do you use: 3.24.21
 - Version of Xournal++: 1.1.0+dev
 - Installation method: GIT

**Describe the bug**
For this bug you need a pressure sensitive pen.
There are cases when a stroke is drawn but the tip of the stylus does not sense any pressure. This can be caused from a bug (#2053) or when a input event is very short in time. In my case this happens a lot when i try to write the dot over the letter 'i'. In 20% of the times the dot will not be recognized.
The issue here is that the stroke will get drawn but with the width of "0.00010757" which is extremely thin. Xournalpp does not display such thin lines at all. However they are in the xopp file and when exported as PDF they will be shown by most of the PDF viewers as a very thin line. (I will add a xopp file)

A quick fix for the problem would be to add a lower limit to the thickness of all strokes. That way the user can see the stroke at least in Xournalpp and erase it if it wasn't on purpose.
For testing purposes I added those lines to StrokeHandler.cpp:71
```
if (Point::NO_PRESSURE != pos.pressure && stroke->getToolType() == STROKE_TOOL_PEN) {
    const double LOWER_LIMIT = 0.2;
    double pressure = pos.pressure;
    if(pressure < LOWER_LIMIT) {
        pressure = LOWER_LIMIT;
    }
    stroke->setLastPressure(pressure * stroke->getWidth());
}
```
And voila no invisible strokes any more. Also all dots over the letter 'i' get recognized now (but they are sometimes really thin tho).

**To Reproduce**
1. Open the xopp file I uploaded.
2. Verify that you see a blank page.
3. Export as PDF and see the thin stroke.

[thin_line.xopp.txt](https://github.com/xournalpp/xournalpp/files/4941368/thin_line.xopp.txt)
[i.xopp.txt](https://github.com/xournalpp/xournalpp/files/4941370/i.xopp.txt)

**Additional**
Adding more pressure steps in general may make a lower pressure limit obsolete, but this has to be tested.

**Comment by rolandlo** [comment]
That makes sense to me. I would have to think harder, if there are any major drawbacks with this idea. Would you mind providing a PR for testing purposes?

**Comment by rolandlo** [comment]
I have examined the problem more closely. There are two different approaches to deal with the problem:

- modify the pressure (like in your PR) when the stroke is drawn by the user
- modify the line width when the stroke is displayed

Pdf-Viewers follow the second approach. We may want to do the same in Xournal++ to get a similar viewing experience. In that case we would have to modify the drawWithPressure() method in the StrokeView class.

**Comment by rolandlo** [comment]
Fixed by #2622
